src/index.js

```diff
@@ -13,17 +13,19 @@ ReactDOM.render(
     <Router>
       <AuthProvider>
         <VideosProvider>
-          <PlaylistModalProvider>
-            <PlaylistsProvider>
-              <LikesProvider>
-                <WatchLaterProvider>
-                  <HistoryProvider>
-                    <App />
-                  </HistoryProvider>
-                </WatchLaterProvider>
-              </LikesProvider>
-            </PlaylistsProvider>
-          </PlaylistModalProvider>
+          <CategoryProvider>
+            <PlaylistModalProvider>
+              <PlaylistsProvider>
+                <LikesProvider>
+                  <WatchLaterProvider>
+                    <HistoryProvider>
+                      <App />
+                    </HistoryProvider>
+                  </WatchLaterProvider>
+                </LikesProvider>
+              </PlaylistsProvider>
+            </PlaylistModalProvider>
+          </CategoryProvider>
         </VideosProvider>
       </AuthProvider>
     </Router>
```

Comment:
Your `Router` should be the innermost child so that it has access to all the providers. It is a good practice to follow.